librz/analysis/var.c

```diff
@@ -618,6 +602,59 @@ static bool var_add_structure_fields_to_list(RzAnalysis *a, RzAnalysisVar *av, R
 	return false;
 }
 
+/**
+ * \brief      Counts the number of var types based on their kind
+ *
+ * \param      fcn   The RzAnalysisFunction to use
+ * \param[in]  kind  The kind of the vars
+ * \param[in]  type  The type of the vars
+ *
+ * \return     The counted number of var types based on their kind
+ */
+RZ_API size_t rz_analysis_var_count(RZ_NONNULL RzAnalysisFunction *fcn, RzAnalysisVarKind kind, RzAnalysisVarType type) {
+	rz_return_val_if_fail(fcn && IS_VALID_VAR_KIND(kind) && type < RZ_ANALYSIS_VAR_TYPE_SIZE, 0);
+
+	size_t count = 0;
+	RzAnalysisVar *var = NULL;
+	RzListIter *iter = NULL;
+	RzList *list = rz_analysis_var_list(fcn, kind);
+	if (!list) {
+		return 0;
+	}
+
+	if (kind == RZ_ANALYSIS_VAR_KIND_REG && type == RZ_ANALYSIS_VAR_TYPE_ARGUMENT) {
+		// all registers types are considered as arguments.
+		count = rz_list_length(list);
+		goto end;
+	}
+
+	rz_list_foreach (list, iter, var) {
+		if (type == RZ_ANALYSIS_VAR_TYPE_LOCAL) {
+			count += var->isarg ? 0 : 1;
+		} else /* if (type == RZ_ANALYSIS_VAR_TYPE_ARGUMENT) */ {
+			count += var->isarg ? 1 : 0;
+		}
+	}
+
+end:
+	rz_list_free(list);
+	return count;
+}
+/**
+ * \brief      Counts the total number of var types
+ *
+ * \param      fcn   The RzAnalysisFunction to use
+ * \param[in]  type  The type of the vars
+ *
+ * @return     The total number of var of a specific type
```

Comment:
By the way, I missed it here - please fix in the next PR:
- Empty line between functions
- `\return` instead of `@return`

Reply:
@wargio